src/lib/components/tabs/QTab.svelte

```diff
@@ -1,86 +1,55 @@
 <script lang="ts">
-  import { onMount } from "svelte";
+  import { getContext, hasContext } from "svelte";
   import { QIcon } from "$lib";
+  import QContext from "$lib/classes/QContext.svelte";
   import { ripple } from "$lib/helpers";
-  import {
-    isActivationKey,
-    isArrowKey,
-    getDirection,
-    getClosestFocusableSibling,
-    isTabKey,
-    getClosestFocusableBlock,
-  } from "$lib/utils";
-  import type { Direction } from "$lib/utils";
+  import { getClosestFocusableBlock, getClosestFocusableSibling } from "$lib/utils/dom";
+  import type { Direction } from "$lib/utils/events";
+  import { getDirection, isActivationKey, isArrowKey, isTabKey } from "$lib/utils/events";
   import { isRouteActive } from "$lib/utils/router";
-  import QContext from "$lib/classes/QContext.svelte";
-  import type { QTabProps } from "./props";
-  import type { QTab, QTabStore } from "./QTabs.svelte";
+  import type { QTabEl } from "./QTabs.svelte";
+  import type { QTabProps, QTabsVariants } from "./props";
 
   type QTabEvent<T> = T & {
-    currentTarget: EventTarget & HTMLElement;
+    currentTarget: EventTarget & QTabEl;
   };
 
-  let { name, to, icon, iconSnippet, children, ...props }: QTabProps = $props();
+  let { name, to, icon, children, ...props }: QTabProps = $props();
 
-  let qTab: QTab;
+  let qTab: QTabEl;
 
-  const ctx = QContext.get<QTabStore>("q-tabs");
+  const tag = $derived(to ? "a" : "button");
 
-  const qTabStore = $derived(ctx?.value);
+  if (!hasContext("QTabsValue")) {
```

Comment:
Can't you check that via `QContext`? It'd be better to have a single "interface" here and not expose the underlying code of `QContext` (so it can be changed at any time).

Reply:
Sure, I can change that in another PR because it would mean changing the API. Unless you want me to change it in this one

Reply:
No, fine for now. :)